manifests/install.pp

```diff
@@ -14,9 +14,6 @@
       ensure => "directory";
   }
 
-  systemd::unit_file {"${nats::service_name}.service":
-    content => epp("nats/systemd_service.epp")
-  }
-
+  Class["nats::install::${nats::service_type}"] ~> Class["nats::service"]
```

Comment:
That's true I suspect it won't probably fine then as is - or do the notify in each of the system specific class